### PR TITLE
feat: Simplify MetricPipeline namespace filter

### PR DESCRIPTION
**Pull request by skhalash**
## Description

Changes proposed in this pull request (what was done and why):

- Change `Enabled` Flag in `MetricPipeline` inputs from Pointer to Boolean. Change `Enabled` to `Disabled` in `otlp` input.
- Make optional `Input` and `Namespace` fields nullable to comply with Kubernetes API conventions: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required 
- Get rid of the `System` option in `MetricPipelineInputNamespaceSelector`
- Update Busola extension and user docs
- Adjust existing E2E tests and add a few new ones

Changes refer to particular issues, PRs or documents:

- https://github.com/kyma-project/telemetry-manager/issues/607

## Traceability
- [x] The PR is linked to a GitHub issue.
- [x] New features have a milestone set.
- [x] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [x] Adjusted the documentation if the change is user-facing.
- [x] The feature is unit-tested
- [x] The feature is e2e-tested

### CODE REVIEW

File: docs/user/04-metrics.md [hunk 1]
```diff
@@ -412,26 +412,7 @@ spec:
...
-      enabled: true
-      namespaces:
-        system: true
-  output:
-    otlp:
-      endpoint:
-        value: https://backend.example.com:4317
-```
-
+Note that the metrics from system Namespaces are excluded by default for the `prometheus`, `runtime`, and `otlp` inputs. However, the metrics from system Namespaces are included by default for the `istio` input.
```
Comment (NHingerl) [inline docs/user/04-metrics.md:415]: [guidelines are changing](https://github.com/kyma-project/kyma/issues/18282#issuecomment-1846626347); "namespace" will be lower case

```suggestion
Note that the metrics from system namespaces are excluded by default for the `prometheus`, `runtime`, and `otlp` inputs. However, the metrics from system namespaces are included by default for the `istio` input.
```

Comment (skhalash) [reply docs/user/04-metrics.md]: Thanks, good to know! I also improved the wording a bit

**Review by NHingerl — DISMISSED** [review]

**Review by NHingerl — APPROVED** [review]

File: apis/telemetry/v1alpha1/metricpipeline_types.go [hunk 2]
```diff
@@ -64,58 +64,57 @@ type MetricPipelineSpec struct {
...
 // MetricPipelineOtlpInput defines the collection of push-based metrics that use the OpenTelemetry protocol.
 type MetricPipelineOtlpInput struct {
-	// If enabled, push-based OTLP metrics are collected. The default is `true`.
-	Enabled *bool `json:"enabled,omitempty"`
-	// Describes whether push-based OTLP metrics from specific Namespaces are selected. System Namespaces are disabled by default.
-	Namespaces MetricPipelineInputNamespaceSelector `json:"namespaces,omitempty"`
+	// If disabled, push-based OTLP metrics are not collected. The default is `false`.
+	Disabled bool `json:"disabled,omitempty"`
+	// Describes whether push-based OTLP metrics from specific Namespaces are selected. System Namespaces are enabled by default.
+	//+optional
```
Comment (shorim) [inline apis/telemetry/v1alpha1/metricpipeline_types.go:107]: Why is the `//+optional` marker added here but not added for the other 3 inputs?

**Comment by shorim** [comment]
In [step 8 in the metrics docs ](https://github.com/kyma-project/telemetry-manager/blob/e36e9df543e2d53ef71eef1239ba1c476eaa9fb6/docs/user/04-metrics.md#step-8-add-filters), there is this one sentence `For example, you can specify the namespaces from which metrics are collected or the namespaces from which metrics are dropped, or choose to collect metrics from all namespaces including the system namespaces kube-system, istio-system and kyma-system`. Can you remove the last part `choose to collect metrics from all namespaces including the system namespaces...`? It was added to advertise the `system` namespace selector. Although that the user can still include the system namespaces using the `include` selector, but this is already explained in the first part of the sentence `you can specify the namespaces from which metrics are collected`.

File: internal/otelcollector/config/metric/gateway/config_builder.go [hunk 3]
```diff
@@ -86,108 +88,127 @@ func makeServiceConfig() config.Service {
...
-	if !*input.Prometheus.Enabled {
+	if !isPrometheusInputEnabled(input) {
 		processors = append(processors, "filter/drop-if-input-source-prometheus")
 	}
-	if !*input.Istio.Enabled {
+	if !isIstioInputEnabled(input) {
 		processors = append(processors, "filter/drop-if-input-source-istio")
 	}
-	if !*input.Otlp.Enabled {
+	if input.Otlp != nil && input.Otlp.Disabled {
```
Comment (shorim) [inline internal/otelcollector/config/metric/gateway/config_builder.go:166]: can we also have a `isOtlpInputDisabled(input telemetryv1alpha1.MetricPipelineInput)` function?

File: internal/otelcollector/config/metric/gateway/config_builder.go [hunk 4]
```diff
@@ -86,108 +88,127 @@ func makeServiceConfig() config.Service {
...
-	if !*input.Prometheus.Enabled {
+	if !isPrometheusInputEnabled(input) {
 		cfg.Processors.DropIfInputSourcePrometheus = makeDropIfInputSourcePrometheusConfig()
 	}
-	if !*input.Istio.Enabled {
+	if !isIstioInputEnabled(input) {
 		cfg.Processors.DropIfInputSourceIstio = makeDropIfInputSourceIstioConfig()
 	}
-	if !*input.Otlp.Enabled {
+	if input.Otlp != nil && input.Otlp.Disabled {
```
Comment (shorim) [inline internal/otelcollector/config/metric/gateway/config_builder.go:110]: can we have a `isOtlpInputDisabled(input telemetryv1alpha1.MetricPipelineInput)` function?

File: internal/otelcollector/config/metric/gateway/config_builder.go [hunk 5]
```diff
@@ -86,108 +88,127 @@ func makeServiceConfig() config.Service {
...
-		processorName := getNamespaceFilterProcessorName(pipeline.Name, metric.InputSourceIstio)
-		cfg.Processors.NamespaceFilters[processorName] = makeFilterByNamespaceIstioInputConfig(pipeline.Spec.Input.Istio.Namespaces)
+	if isIstioInputEnabled(input) && shouldFilterByNamespace(input.Istio.Namespaces) {
+		processorID := makeNamespaceFilterID(pipeline.Name, metric.InputSourceIstio)
+		cfg.Processors.NamespaceFilters[processorID] = makeFilterByNamespaceIstioInputConfig(pipeline.Spec.Input.Istio.Namespaces)
 	}
-	if shouldFilterByNamespace(input.Otlp.Enabled, input.Otlp.Namespaces) {
-		processorName := getNamespaceFilterProcessorName(pipeline.Name, metric.InputSourceOtlp)
-		cfg.Processors.NamespaceFilters[processorName] = makeFilterByNamespaceOtlpInputConfig(pipeline.Spec.Input.Otlp.Namespaces)
+	if input.Otlp != nil && !input.Otlp.Disabled && shouldFilterByNamespace(input.Otlp.Namespaces) {
```
Comment (shorim) [inline internal/otelcollector/config/metric/gateway/config_builder.go:133]: After defining the `isOtlpInputDisabled(input telemetryv1alpha1.MetricPipelineInput)` function we can check here if `!isOtlpInputDisabled(input) && shouldFilterByNamespace(input.Otlp.Namespaces)`

File: internal/otelcollector/config/metric/gateway/config_builder.go (hunk 3 above)
Comment (shorim) [inline internal/otelcollector/config/metric/gateway/config_builder.go:166]: also here we can do the check using the `isOtlpInputDisabled(input telemetryv1alpha1.MetricPipelineInput)` function

File: internal/otelcollector/config/metric/gateway/config_builder.go [hunk 6]
```diff
@@ -86,108 +88,127 @@ func makeServiceConfig() config.Service {
...
-	if shouldFilterByNamespace(input.Istio.Enabled, input.Istio.Namespaces) {
-		processorName := getNamespaceFilterProcessorName(pipeline.Name, metric.InputSourceIstio)
-		processors = append(processors, processorName)
+	if isIstioInputEnabled(input) && shouldFilterByNamespace(input.Istio.Namespaces) {
+		processors = append(processors, makeNamespaceFilterID(pipeline.Name, metric.InputSourceIstio))
 	}
-	if shouldFilterByNamespace(input.Otlp.Enabled, input.Otlp.Namespaces) {
-		processorName := getNamespaceFilterProcessorName(pipeline.Name, metric.InputSourceOtlp)
-		processors = append(processors, processorName)
+	if input.Otlp != nil && !input.Otlp.Disabled && shouldFilterByNamespace(input.Otlp.Namespaces) {
```
Comment (shorim) [inline internal/otelcollector/config/metric/gateway/config_builder.go:179]: also here we can do the check if `!isOtlpInputDisabled(input) && shouldFilterByNamespace(input.Otlp.Namespaces)`

File: apis/telemetry/v1alpha1/metricpipeline_types.go (hunk 2 above)
Comment (skhalash) [reply apis/telemetry/v1alpha1/metricpipeline_types.go:107]: Agree, will add it to the rest of inputs and namespaces

**Review by NHingerl — APPROVED** [review]

File: internal/otelcollector/config/metric/gateway/config_builder.go [hunk 7]
```diff
@@ -212,3 +212,8 @@ func isRuntimeInputEnabled(input telemetryv1alpha1.MetricPipelineInput) bool {
 func isIstioInputEnabled(input telemetryv1alpha1.MetricPipelineInput) bool {
 	return input.Istio != nil && input.Istio.Enabled
 }
+
+// isOtlpInputEnabled
```
Comment (shorim) [inline internal/otelcollector/config/metric/gateway/config_builder.go:216]: I think that this comment was probably unintentionally committed, right?

Comment (skhalash) [reply internal/otelcollector/config/metric/gateway/config_builder.go:216]: True, will remove it!

**Review by grego952 — APPROVED** [review]

**Review by shorim — APPROVED** [review]

**Pull request merged**